### PR TITLE
Refactor: use `format-args-capture` and remove unnecessary nested if blocks in some parts of `rust_passes`

### PR DESCRIPTION
**Pull request by TaKO8Ki**

### CODE REVIEW

**Comment by bugadani** [comment]
@lcnr 

Is it a good idea to approve PRs like this while a similar one might potentially being reverted? https://github.com/rust-lang/rust/pull/95062

**Comment by lcnr** [comment]
didn't know about the revert, considering that the reason was a perf regression

@bors try @rust-timer queue

**Comment by cronlabspl** [comment]
Diving SOLLO shitasses

File: compiler/rustc_passes/src/intrinsicck.rs [hunk 1]
```diff
@@ -509,14 +505,12 @@ impl<'tcx> Visitor<'tcx> for ExprVisitor<'tcx> {
...
                 let res = self.typeck_results.qpath_res(qpath, expr.hir_id);
-                if let Res::Def(DefKind::Fn, did) = res {
-                    if self.def_id_is_transmute(did) {
-                        let typ = self.typeck_results.node_type(expr.hir_id);
-                        let sig = typ.fn_sig(self.tcx);
-                        let from = sig.inputs().skip_binder()[0];
-                        let to = sig.output().skip_binder();
-                        self.check_transmute(expr.span, from, to);
-                    }
+                if let Res::Def(DefKind::Fn, did) = res && self.def_id_is_transmute(did) {
```
Comment (joshtriplett) [inline compiler/rustc_passes/src/intrinsicck.rs:508]: Please break before the `&&` here; that seems more readable.

Comment (TaKO8Ki) [reply compiler/rustc_passes/src/intrinsicck.rs]: I fixed it :)

**Comment by cronlabspl** [comment]
Rust should become extint ASAP.

**Pull request merged**